### PR TITLE
[Data] Move release test config to separate file

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

Storing all of the Data release test configs in a separate file makes it easier to maintain the Data release tests.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/release_data_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,910 @@
+###############
```
Comment (omatthew98) [inline release/release_data_tests.yaml:1]: Just to confirm this is just a copy paste from the existing `release_tests.yaml`? Assuming I don't need to review any of this.

**Review by omatthew98 — APPROVED** [review]
Lgtm!

File: release/release_data_tests.yaml (hunk 1 above)
Comment (bveeramani) [reply release/release_data_tests.yaml:1]: Yeah. Copy and paste of the "Batch inference" and "Data tests" sections

**Review by can-anyscale — APPROVED** [review]
Very nice. I think you might or might not need to update on the turbo side as well.

**Pull request merged**